CMake/CMakeLibs.cmake

```diff
@@ -68,50 +68,64 @@ macro(ADD_OSQUERY_LINK IS_CORE LINK)
   endif()
 endmacro(ADD_OSQUERY_LINK)
 
-macro(ADD_OSQUERY_LINK_INTERNAL LINK LINK_PATHS LINK_SET)
-  set(LINK_PATHS_RELATIVE
-    "${BUILD_DEPS}/lib"
-    "${CMAKE_BUILD_DIR}/third-party/*/lib"
-    ${LINK_PATHS}
+if(WIN32)
+  foreach(DEP_DIR ${WINDOWS_DEP_LIST})
+    file(GLOB_RECURSE WINDOWS_LOCAL_LIB_DIRS LIST_DIRECTORIES false "${DEP_DIR}/*.lib")
+    foreach(LIB_FILE ${WINDOWS_LOCAL_LIB_DIRS})
+      get_filename_component(LIB_DIR ${LIB_FILE} DIRECTORY)
+      list(APPEND OS_LIB_DIRS ${LIB_DIR})
+    endforeach()
+  endforeach()
+  list(REMOVE_DUPLICATES OS_LIB_DIRS)
+  foreach(LINK_DIR ${OS_LIB_DIRS})
+    link_directories(${LINK_DIR})
```

Comment:
Thoughts: move this whole block to the main CMakeLists.txt (it doesn't belong in CMakeLibs.cmake) and then call `link_directories(${OS_LIB_DIRS})` instead of doing the for loop?